js/private/test/shellcheck_launcher.sh

```diff
@@ -136,116 +257,110 @@ else
-    if [ "${JS_BINARY__VERBOSE:-}" ]; then
-        echo "${LOG_PREFIX}: changing directory to user specified package $JS_BINARY__CHDIR" >&2
-    fi
+    logf_debug "changing directory to user specified package %s" "$JS_BINARY__CHDIR"
     cd "$JS_BINARY__CHDIR"
 fi
 
-if [ "${JS_BINARY__CAPTURE_STDOUT:-}" ]; then
-    STDOUT_CAPTURE="$JS_BINARY__CAPTURE_STDOUT"
-fi
-
-if [ "${JS_BINARY__CAPTURE_STDERR:-}" ]; then
-    STDERR_CAPTURE="$JS_BINARY__CAPTURE_STDERR"
-fi
-
-if [ "${JS_BINARY__SILENT_ON_SUCCESS:-}" ]; then
-  if [ -z "${STDOUT_CAPTURE:-}" ]; then
-    STDOUT_CAPTURE_IS_NOT_AN_OUTPUT=true
-    STDOUT_CAPTURE=$(mktemp)
-  fi
-  if [ -z "${STDERR_CAPTURE:-}" ]; then
-    STDERR_CAPTURE_IS_NOT_AN_OUTPUT=true
-    STDERR_CAPTURE=$(mktemp)
-  fi
-fi
-
-# Bash does not forward termination signals to any child process when
-# running in docker so need to manually trap and forward the signals
-_term() {
-  kill -TERM "${child}" 2>/dev/null
-}
-
-_int() {
-  kill -INT "${child}" 2>/dev/null
-}
-
-_exit() {
-  EXIT_CODE=$?
-
-  if [ "$EXIT_CODE" != 0 ]; then
-    if [[ ${STDOUT_CAPTURE_IS_NOT_AN_OUTPUT:-} == true ]]; then
-      cat "$STDOUT_CAPTURE"
-      rm "$STDOUT_CAPTURE"
-    fi
-    if [[ ${STDERR_CAPTURE_IS_NOT_AN_OUTPUT:-} == true ]]; then
-      cat "$STDERR_CAPTURE"
-      rm "$STDERR_CAPTURE"
-    fi
-  fi
-
-  exit $EXIT_CODE
-}
-
 NODE_OPTIONS=()
 
 
 ARGS=()
 ALL_ARGS=( "$@")
 for ARG in ${ALL_ARGS[@]+"${ALL_ARGS[@]}"}; do
-  case "$ARG" in
-    # Let users pass through arguments to node itself
-    --node_options=*) NODE_OPTIONS+=( "${ARG#--node_options=}" ) ;;
-    # Remaining argv is collected to pass to the program
-    *) ARGS+=( "$ARG" )
-  esac
+    case "$ARG" in
+        # Let users pass through arguments to node itself
+        --node_options=*) NODE_OPTIONS+=( "${ARG#--node_options=}" ) ;;
+        # Remaining argv is collected to pass to the program
+        *) ARGS+=( "$ARG" )
+    esac
 done
 
 # Put bazel managed node on the path
 PATH="$(dirname "$node"):$PATH"
 export PATH
 
+# Debug logs
+if [ "${JS_BINARY__LOG_DEBUG:-}" ]; then
+    if [ "${BAZEL_BINDIR:-}" ]; then
+        logf_debug "BAZEL_BINDIR %s" "${BAZEL_BINDIR:-}"
+    fi
+    if [ "${BAZEL_TARGET_CPU:-}" ]; then
+        logf_debug "BAZEL_TARGET_CPU %s" "${BAZEL_TARGET_CPU:-}"
+    fi
+    if [ "${BAZEL_COMPILATION_MODE:-}" ]; then
+        logf_debug "BAZEL_COMPILATION_MODE %s" "${BAZEL_COMPILATION_MODE:-}"
+    fi
+    if [ "${BAZEL_WORKSPACE:-}" ]; then
+        logf_debug "BAZEL_WORKSPACE %s" "${BAZEL_WORKSPACE:-}"
+    fi
+    if [ "${BAZEL_BUILD_FILE_PATH:-}" ]; then
+        logf_debug "BAZEL_BUILD_FILE_PATH %s" "${BAZEL_BUILD_FILE_PATH:-}"
+    fi
+    if [ "${BAZEL_INFO_FILE:-}" ]; then
+        logf_debug "BAZEL_INFO_FILE %s" "${BAZEL_INFO_FILE:-}"
+    fi
+    if [ "${BAZEL_VERSION_FILE:-}" ]; then
+        logf_debug "BAZEL_VERSION_FILE %s" "${BAZEL_VERSION_FILE:-}"
+    fi
+    logf_debug "binary target BINDIR %s" "${JS_BINARY__BINDIR:-}"
```

Comment:
would it be shorter to logf_debug once with a multiline argument?

Reply:
the ordering of the `%s` statements becomes unruly with a single call